### DISCUSSION TITLE
Missing workflow_dispatch at the Action tab

### DISCUSSION THREAD

**Discussion by diegobernardes** [community/community#25219] [Actions]
<p>I was doing some tests with the <code>workflow_dispatch</code> and it was working great until I did a force push into master (it’s a private and not used by others repository) to move the changes from master to a pull request.<br>
One of those changes was the <code>workflow_dispatch</code>. After I did this the workflow doesn’t shows up anymore at the Action tab.</p>
<p>GitHub got into an invalid state? What I can do to solve this problem?</p>
<p>I tried to delete the file and then create using the graphic interface, but I got the same result.</p>

**Comment by Simran-B** [comment]
<p>The workflow file must be committed in the default branch (master/main) if you want to use the <code>workflow_dispatch</code> event. Is it still present there or only on a feature branch?</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow-on-github" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow-on-github" target="_blank" rel="noopener nofollow ugc">Manually running a workflow - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by diegobernardes** [comment]
<p>Yes, the file is at the default branch and still doesn’t show to me on the actions tab. It’s not the first time this happens to me. Looks like GitHub get into an invalid state and after that, I can’t use the manual workflow at the repository anymore.</p>
<p>It’s not like I can’t see the <code>Run workflow</code> button, I can’t see the whole workflow. The other workflows that I have at the repository are there and working fine.</p>

**Comment by diegobernardes** [comment, marked as answer]
<p>It’s a bug with GitHub, I just confirmed. I changed the name of the files and the name of the workflow and now it’s showing for me.</p>

**Reply by gvegayon** [reply]
May 2024 and I see the same issue. Not sure why but renaming the workflow makes the button show up.

**Reply by lucasp-plore** [reply]
August 2024 and I just faced the same issue...

**Reply by DanielStoica85** [reply]
Were you guys able to fix it (in 2024)?

**Reply by gopinath-ahold** [reply]
End of 2024. Still not fixed

**Reply by Sukhtaaj** [reply]
Last day of 2024; Same state.

**Reply by oskarsword** [reply]
This is still an issue just putting this here

**Reply by derekmckinnon** [reply]
Just hit this annoying issue 😭

**Reply by trucpd** [reply]
2026

**Comment by leafney** [comment]
<p>Thanks. I had the same problem with not displaying the workflow button on the page. After changing the file name, it works.</p>
<pre><code class="lang-auto">Changes to be committed:
  (use "git restore --staged &lt;file&gt;..." to unstage)
        renamed:    .github/workflows/develop.yml -&gt; .github/workflows/build.yml

</code></pre>

**Comment by mvlasovatl** [comment]
<p>May we get a notification, when this bug is fixed? It’s quite annoying. I faced it as well.</p>

**Comment by andymckay** [comment]
<p>This has been fixed, if you are still seeing a problem, please let us know where.</p>

**Reply by duaneking** [reply]
It has not been fixed.

**Comment by mvlasovatl** [comment]
<p>Hey <a class="mention" href="/andymckay">@andymckay</a> !<br>
It happened just today in <a href="https://github.com/mvlasovatl/atlassian-slack-integration-server" rel="noopener nofollow ugc">my private repository</a>. I have pushed there a copy of <a href="https://github.com/atlassian-labs/atlassian-slack-integration-server" rel="noopener nofollow ugc">the public repository</a> with already existing workflows. <a href="https://github.com/atlassian-labs/atlassian-slack-integration-server/blob/59ba2c978c15098cce54ced17463abcdf8628751/.github/workflows/release.yml" rel="noopener nofollow ugc">The workflow that was supposed to be triggered manually only</a> wasn’t detected by Github and therefore didn’t show up in <strong>Actions</strong> tab. So, I wasn’t able to trigger it. Neither via UI, nor via REST API. I had to rename a workflow file in <code>master</code> to make Github notice that workflow and let me run it eventually. Renaming a workflow in a branch didn’t help.</p>

**Comment by andymckay** [comment]
<p>Thank you, I’ll file an issue internally and the team will take a look.</p>

**Reply by duaneking** [reply]
It still has not been fixed.

**Comment by lupine86** [comment]
<p>I was able to reproduce what you describe.  I didn’t need to rename the file, but instead just added a comment to the file and noticed the workflow was created.</p>

**Reply by beikeni** [reply]
Same exact issue here

**Comment by nbdd0121** [comment]
<p>Still couldn’t get it to work in a private clone of a private repository. I’ve tried adding a comment as <a class="mention" href="/lupine86">@lupine86</a> suggested but the button still didn’t appear.</p>

**Comment by modullar** [comment]
<p>It is not working for me either! no button for the workflow is being shown.</p>

**Comment by pdolinic** [comment]
<p>I can confirm renaming workflow from the the orignal.yml  to fixed.yml fixed it for me as well. It is then also possible to rename to the orginal.yml to have it work afterwards.</p>

**Comment by abhinavan1812** [comment]
Hi @andymckay,

Seems like this issue is still exist.
If not please guide me to the possible solution for this problem.

Thanks.

**Comment by srinivasan-getstan** [comment]
Hi @andymckay,

The issue still exists. Any update?

**Comment by andymckay** [comment]
@srinivasan-getstan, @abhinavan1812 I no longer work at GitHub, so am not able to provide much insight, sorry.

**Comment by reza305z** [comment]
Hi everyone

Same problem here (private repo)!

**Comment by Kareemuddin-abjayon12** [comment]
Not able to see the Run Manual Workflow option even though if i give the workflow_dispatch in my yml file as attached in the below screenshot,
![YML file screenshot](https://user-images.githubusercontent.com/121866342/211824782-a6a739a8-dcba-427c-bb7c-80280ee31913.JPG)

Let me know if iam missing anything here!!

**Comment by your-diary** [comment]
Edit: Double-cheking the default branch fixed the problem. I thought the default branch was `master` but for some reason it was set to an inactive feature branch.

This very simple yaml doesn't work.

- GUI button not shown

- `gh workflow run` gives `could not create workflow dispatch event: HTTP 422: Workflow does not have 'workflow_dispatch' trigger`

- renaming the name of the yaml file didn't fix the problem

```yaml
name: hello
on:
    push:
    workflow_dispatch:
jobs:
    hello_world:
        runs-on: ubuntu-latest
        steps:
            - run: echo "Hello, world!"
```

**Comment by Leehaeun0** [comment]
I changed the file name but it didn't work, but changing the name of the action solved it!

```
name: hello-world // change this
```

**Comment by kami619** [comment]
same thing happens for me as well on my private_repo and its quite annoying.

**Comment by aries1980** [comment]
It is still happening `workflow_dispatch`. The fix was to rename the filels but the workflow that has no longer a file can't be deleted.

**Comment by VladyslavKurmaz** [comment]
Almost three years, bug is still there.
* Dispatch button press says that workflow execution was successfully requested but nothing happens
* Github doesn't display workflow title correctly

Workflow is in main branch
```
name: "Dispatch"

on:
  workflow_dispatch:
    inputs: 
      environment:
        type: string
        description: Environment
```
<img width="274" alt="Screenshot 2023-08-08 at 09 48 34" src="https://github.com/community/community/assets/1977961/7b70f001-1807-47d5-bb79-07352331d5c5">

**Reply by alexkates** [reply]
Running into this as well, exactly how @VladyslavKurmaz described it above.

I added a comment to the top of any missing workflow files and pushed to the default branch. This fixed the issue and they are now showing in the Actions tab in the Github UI.

**Reply by gezichenshan** [reply]
Doen't work with renaming and commenting on a new branch.

**Comment by fsadykov** [comment]
Stil facing this issue. Now able to trigger the actions from the UI manually 

```
name: The Docker Build Pipeline

on:
  workflow_dispatch:
  push:
    branches:
    - 'dev-*'
    - 'qa-*'
    - 'master'
    - 'v*'

jobs:
  common-actions:
    uses: fuchicorp/common_actions/.github/workflows/build.yaml@feature/master
    secrets: inherit
 
```

**Comment by JamesCalleja** [comment]
I had to change the default branch from main to master to resolve this. Same yaml file for the workflow .

**Comment by mikesmithgh** [comment]
I ran into this issue today. After renaming the file, renaming the workflow name, and merging to main, the `Run workflow` button is available. See https://github.com/mikesmithgh/kitty-scrollback.nvim/pull/102

**Comment by mlutsoucs** [comment]
Unfortunately, 2024  problem still exist.
It's absolutely random. Some actions not fixed even after renaming/recreation, some after several pushes and some works as expected  immediately after creation...

**Comment by jhchill666** [comment]
This is driving me nuts.  Adding new `workflow_dispatch` workflow, does just not pick up.  If I copy contents into an existing (working) `workflow_dispatch` it runs perfectly fine.  If I rename this workflow.yml to the non-working name, it stops working.

Can't be more simple than this:
```
name: CI Sandbox

on:
  workflow_dispatch:

jobs:
  Sandbox:
    uses: ./.github/workflows/deploy-env.yml
    with:
      env: sandbox
    secrets: inherit
```

**Comment by atipi** [comment]
Hi, I ran to this problem today and changing file extension from .yml to .yaml didn't help. I even tired to change back and forth but no luck. And I tried to see UI from own branch, not main branch for testing purpose.

**Comment by roshan-mosaicai** [comment]
Whats the fix for this issue, tried all the suggested steps in this thread

**Reply by roshan-mosaicai** [reply]
Just to close loop on this, As suggested by @mikesmithgh, once merged into main branch, i was able to see the workflow.

**Comment by armenavanesi** [comment]
How is there not a fix for this issue yet? I've created a workflow WITH a workflow_dispatch and any time I select a tag or branch I get:
`Workflow does not exist or does not have a workflow_dispatch trigger in this tag.`
I've even copy pasted a workflow that DOES work and thats what it says, even though they are identical.

**Comment by clarkmcc** [comment]
I am surprised this is still an issue. It's not always possible to just move the pipeline to the default branch. We run our own fork of armbian/build and operate off of a versioned branch which we then apply our own patches to as well as our own build pipeline. Right now we can't run that build pipeline unless we push that to master the default branch. But these versioned branches _should never_ be merged to the default branch.

Please fix :)

**Reply by Steve-Glass** [reply]
@clarkmcc At present, the workflow file must be committed in the default branch. Given the scenario described above, are you asking for an enhancement that allows you to manually trigger workflows from any branch?

**Reply by duaneking** [reply]
My default expectation would be that for accessibility and functionality, that I could trigger a workflow for any branch or changeset.

**Comment by lucaovieira** [comment]
I'm facing this problem now, I have to change the default branch for it to work, it's amazing how the problem is still active and no one cares.

**Comment by Nacosia** [comment]
bro, it's really really bad when you think it's your problem and you spend all day troubleshooting, then find out it's a bug.    :(

**Comment by Steve-Glass** [comment]
👋 Chiming in from the Actions product team. I'll see if there are any existing internal issues to address this, along with trying to reproduce it myself.

**Reply by Steve-Glass** [reply]
I can't find any existing internal issues and have added this to our backlog of UX improvements for consideration. I will circle back to this thread once planning is done this month. I can't guarantee it will be prioritized immediately given existing objectives. However, I will try and confirm when we expect to tackle this.

**Comment by danonorato** [comment]
i can confirm this is still broken today from the gh client

gh workflow run ch-cluster-adhoc-loader.yml -f deliveryDate=2024-10-03-03 -f adhocHost=foobar --ref do/ch-cluster-adhoc-loader

It will not trigger the workflow.   I understand if I add my merge my branch into the default branch it will work.  I have other workflow in my default branch.  HOWEVER,  how are we expected to test dispatch workflows?  Iterate in the default branch??????  can't be the case right?

**Comment by NinoScript** [comment]
Please add a branch selector to the Actions tab. This will allow me to view the Actions I'm working on and iterate on them before merging to the default branch.

**Comment by EmiliaKlassonZendr** [comment]
Can recommend the workaround described here: 
https://thomaslevesque.com/2024/04/25/running-a-github-actions-workflow-that-doesnt-exist-yet-on-the-default-branch/. 

Makes it possible to deploy from a non-default branch.

**Reply by brekkfast** [reply]
This works for my use case, thanks for sharing

**Comment by VillagerTom** [comment]
I come across this issue today when I'm writing a new workflow file.
All my workflow files have the same fields as below:
```yaml
on:
  workflow_dispatch:
  push:
    branches:
      - 'main'
    paths-ignore:
      - '**.md'
      - '**.txt'
      - '.github/**'
      - '.idea/**'
      - '!.github/workflows/${workflow_filename}'
```
But only one workflow has the `Run workflow` button!

**Comment by aniagarw-cc** [comment]
weirdly my workflow file in non-default branch is working when called from ghcli but don't see the option to run it from UI or port.io

**Comment by krakjn** [comment]
### TL;DR must commit workflow files back to `default` branch. 
That's how they show up on GUI. I was working on a side branch and they were not showing up. A workflow is a per file basis. 
<img width="1071" alt="image" src="https://github.com/user-attachments/assets/dcafa16c-acc9-4898-b102-afdc171a7a58" />

**Reply by anthops** [reply]
Ah damn that sucks

**Comment by mjohnsonengr** [comment]
I wonder if I ended up in this state by forking a repo that had actions.

**Comment by yourcharlie23** [comment]
Bug still exist, workflow wont show up until you rename/delete and put back the the yml file. This bug always happen if the repo is forked/cloned from a repo/template that has action. It wont automatically list the workflow until you delete/rename the file... Editing the file may make it appear but it wont trigger, no history of running at all.

**Reply by vikash2310** [reply]
It seems this issue still isn't fully resolved. If you're not seeing the workflow trigger button, make sure you've copied your workflow file to the default branch (usually main or master).

Once the file is present in the default branch, the "Run workflow" button should appear in the GitHub Actions tab for that workflow.

Here’s a minimal example that should work:

```
on:
  workflow_dispatch:
    inputs:
      environment:
        description: 'Environment to deploy to'
        required: true
        default: 'dev'
        type: choice
        options:
          - dev
```

**Comment by ijiratth** [comment]
Coming across this bug today as well. The workflow is pushed to default branch directly along with other stuff (1st commit) when the repo first published. The workflow doesn't appear when default branch is named 'master'. It appears after editting the workflow and commit again.

When default branch is named 'main'. It appears as normal from 1st commit.

**Comment by vikash2310** [comment]
It seems this issue still isn't fully resolved. If you're not seeing the workflow trigger button, make sure you've copied your workflow file to the default branch (usually main or master).

Once the file is present in the default branch, the "Run workflow" button should appear in the GitHub Actions tab for that workflow.

Here’s a minimal example that should work:

```
on:
  workflow_dispatch:
    inputs:
      environment:
        description: 'Environment to deploy to'
        required: true
        default: 'dev'
        type: choice
        options:
          - dev
```

**Comment by duaneking** [comment]
This defect still exists and it is not appropriate to expect it to be checked into the main branch as a resolution.

My default expectation for accessibility and functionality is that if I've given this workflow_dispatch tag to it then it's always going to show up and I have the option of running it against any branch or change set that I request.

**Comment by mandrzejewski-iceye** [comment]
Is this going to get fixed ever? You want the users to merge into the master a work in progress code just to make the workflow appear in the Actions tab? That is super weird.

**Comment by hz3866** [comment]
this bug appears again.

"""
name: Publish To PYPI ✋

permissions:
  actions: write
  contents: read
  security-events: write


on:
  workflow_dispatch:
  
  """

no action button

**Comment by spperforce** [comment]
this bug reappeared again
